### ISSUE TITLE
Move the React-Toolkit into this Styleguide repo

### ISSUE THREAD

**Issue by tomatau** [sky-uk/nowtv-react-toolkit#100]
Alright, so this idea might be completely undoing what's already been done... but just go with me here...

**Problems we have right now**

- **Updating the styleguide and toolkit is a huge pain**. 
  - Especially for breaking changes, there's PRs all over the place
  - Things going in and out of sync it's easy to forget to update both styleguide and toolkit at the same time
  - Communicating changes that refer to both projects is also a bit of a hassle.
  - Double release notes, double npm publishes, double docs, double tests (when we do some css regression) it's too easy to forget one of these things.
- **Where to put things** should the toolkit be requiring the styles? Should the styleguide have the fonts? Should the toolkit have icons? With both repos, it's unclear and very subjective about where assets should be living.
- **Accommodating everyone twice** with so many ways to use the combination of the two projects in your app (cdn, import the dist compiled output, import src with various compiler config settings).. it's getting fairly hairy to accommodate all of these options in both projects. And do we even need to accommodate all of these approaches? E.g. Letting apps import the full dist means they're often requiring a bunch of css or javascript that they'll never use.

I'm suggesting we just move all the toolkit components into the styleguide. One place to update. One place to put things. One set of scripts, tests, compilation and documents to maintain.

**New Problems**

- **consumers of the styleguide don't always need the react components** - So what? They don't need to require them.
- **importing it all would be very large** - So we should discourage this 'import it all' approach, and instead encourage consumers to only import what they need instead.
- **Why only React? Should we then support other component types such as angular?** - Consumers might be using the styleguide for angular directives or whatever... they already will need to update those directives when we do styleguide breaking changes anyway... So if those consumers see a benefit to putting them in here... seems ok to me! It could even encourage other teams to get more involved in the global styleguide... There are other downsides to this but I think the benefits greatly outweigh the cons.
- **Other problems?** - You tell me.

**Comment by peter-mouland** [comment]
the projects are very much tied together, and there is interest in creating small react-components to wrap in puts and buttons, which will tie them even closer. So actually makes sense!

Does it raise the point of do we want to split the components into separate repos which contain styles and reactJS? ie. n-accordion component and n-layout (prefix to standfor nowtv or whatever).   This might help reduce breaking changes for the whole toolkit when its only the accordion that changed.

**Comment by tomatau** [comment]
I think lots of little repos for each component would be an even bigger pain. Especially when we want to do a bigger update such as branding in a year from now or something.

Also having each project requiring each component separately would be a tension point too.

I think it's simple enough to just expose each component separately from within one repo.

**Comment by peter-mouland** [comment]
I've seen it done in the past when toolkit type projects get too big, but maybe we're not at that point yet.

**Comment by yhzgirl** [comment]
I can see the benefits of having these things in one repo, there's such a dependancy between the two. If you don't update both and they become out of sync it becomes confusing as to where a problem is. Having the code together would make it easier to update the code as there would be less steps to make sure everything was set up correctly.

**Comment by alexflorisca** [comment]
:+1: I'm all for this too, it's a great idea. If you structure it well, it won't get too big. Bumping version numbers often is only an aesthetic problem and makes total sense to have one 'toolkit' or 'styleguide' or whatever we call it repository. Not sure why we didn't do this sooner really.

And it would future proof it when React fades away and people start building stuff in web components or whatever by having a structure like this, without breaking anyones stuff:

```
/styleguide
/react-components
/web-components
/angular-components
/etc...
```

**Comment by bentidy** [comment]
:+1:

We also find it a pain, we're often building the react components first and then having to port them over to the styleguide. It's an added complexity to have them separated and we don't see much of a benefit to it.

Should definitely check with sales and my account though. Maybe we can bring it up in the next toolkit meeting?

@ldabiralai @nwc1991

**Comment by rogerfranklin2** [comment]
:+1:

**Comment by ariley90** [comment]
So I think this is a good idea using a structure suggested by @alexflorisca above as the current process is very difficult to keep in sync with.

We split it out to stop people just building the components in react but I don't think that is a problem now as everyone seems to agree that a clear seperation between styles and behaviour is useful but whether we split that be repo or by folder structure I think is okay. 1 point to raise would be that if components started to be built using things other than react then we might end up with some bloat as something like angular would potentially add different tooling but we could tackle that if it happens.

My final point would be is now the best time to do this? We have all felt the pain of constant changes over the last few weeks. Do we want to identify the difference between fixes/new features and optimisations? This is probably a topic to bring up in the global catch up.

**Comment by alexjfno1** [comment]
:+1: Makes sense to me as well.

@peter-mouland On your point about the toolkit getting to big. We should only be putting common stuff into the toolkit. I think that if we get to a point where we are putting carousels and 4 types of accordion into it we should have a conversation about weather they should live in the styleguide. From my point of view the styleguide should just contain the core elements and more complex compoents should possibly have their own repo. But we're not there yet so lets not worry about that now.

**Comment by nowarkdev** [comment]
:+1: I think this sounds like a good idea as we have had similar issues with waiting on pull requests between both repos. Even if further down the line we feel it could be split back out due to size, that should be relatively easy to do if we follow a structure similar to that above.

@roryp2

**Comment by peter-mouland** [comment]
done https://github.com/sky-uk/nowtv-web-toolkit
